java/vector/src/test/java/org/apache/arrow/vector/TestValueVector.java

```diff
@@ -1394,6 +1394,35 @@ public void testFillEmptiesNotOverfill() {
     }
   }
 
+  @Test
+  public void testSetSafeWithArrowBuf() {
+    final int numValues = BaseFixedWidthVector.INITIAL_VALUE_ALLOCATION * 2;
+
+    try (
+        final VarCharVector fromVector = newVector(VarCharVector.class, EMPTY_SCHEMA_PATH,
+            MinorType.VARCHAR, allocator);
+        final VarCharVector toVector = newVector(VarCharVector.class, EMPTY_SCHEMA_PATH,
+            MinorType.VARCHAR, allocator)) {
+      fromVector.setInitialCapacity(numValues);
+      fromVector.allocateNew();
+      for (int i = 0; i < numValues; ++i) {
+        fromVector.setSafe(i, "hello world".getBytes(), 0, 11);
+      }
+      fromVector.setValueCount(numValues);
+      ArrowBuf fromDataBuffer = fromVector.getDataBuffer();
+      assertEquals(BaseAllocator.nextPowerOfTwo(numValues * 11), fromDataBuffer.capacity());
+
+      toVector.setInitialCapacity(numValues);
+      toVector.allocateNew();
+      for (int i = 0; i < numValues; i++) {
+        int start = fromVector.getstartOffset(i);
+        int end = fromVector.getstartOffset(i + 1);
+        toVector.setSafe(i, 1, start, end, fromDataBuffer);
```

Comment:
making this a named constant or putting in comments what the value represents might make the test easier to read.

Reply:
done